### PR TITLE
Enabled warnings-as-errors for our Windows builds

### PR DESCRIPTION
**Pull request by bobsummerwill**
All outstanding Windows warnings were fixed or suppressed in previous commits.
Also split the global warning suppressions in webthree-helpers onto a line each, for ease of comprehension.
We have way too many of them, and some may be masking real issues.
This commit is a partial fix for https://github.com/ethereum/webthree-umbrella/issues/219.

### CODE REVIEW

**Comment by bobsummerwill** [comment]
Result masked by existing break.   Let's get that cleared first, and then get this merged after that, if it shows as clean.

File: cmake/EthCompilerSettings.cmake [hunk 1]
```diff
@@ -36,21 +36,21 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
...
-	# disable qualifier applied to function type has no meaning; ignored (4180)
-	# disable C++ exception specification ignored except to indicate a function is not __declspec(nothrow) (4290)
-	# disable conversion from 'type1' to 'type2', possible loss of data (4244)
-	# disable forcing value to bool 'true' or 'false' (performance warning) (4800)
-	# disable warning C4535: calling _set_se_translator() requires /EHa (for boost tests)
-	# declare Windows Vista API requirement
-	# undefine windows.h MAX && MIN macros cause it cause conflicts with std::min && std::max functions
-	# define miniupnp static library
-	add_compile_options(/MP /EHsc /wd4068 /wd4996 /wd4503 /wd4267 /wd4180 /wd4290 /wd4244 /wd4800 -D_WIN32_WINNT=0x0600 /DNOMINMAX /DMINIUPNP_STATICLIB)
+    add_compile_options(/MP)                      # enable parallel compilation
```
Comment (chriseth) [inline cmake/EthCompilerSettings.cmake]: Tab

**Comment by chriseth** [comment]
Build error on windows: Unreferenced local variable in EthereumHost.cpp(315)

File: cmake/EthCompilerSettings.cmake (hunk 1 above)
Comment (bobsummerwill) [inline cmake/EthCompilerSettings.cmake]: Use tabs for the formatting on the right-hand-side, you mean?

**Pull request closed without merging**